Source/projections/Builder/ProjectionClassBuilder.ts

```diff
@@ -0,0 +1,161 @@
+            return;
+        }
+
+        this.ThrowIfReadModelAlreadyRegistered(ProjectionDecoratedTypes.types);
+
+        logger.debug(`Building projection ${decoratedType.projectionId} processing events in scope ${decoratedType.scopeId} from type ${this._projectionType.name}`);
+
+        const methods = OnDecoratedMethods.methodsPerProjection.get(this._projectionType);
+        if (methods === undefined) {
+            logger.warn(`There are no projection methods to register in projection ${this._projectionType.name}. An projection must to be decorated with @${onDecorator.name}`);
+            return;
+        }
+        const events = new EventTypeMap<[ProjectionCallback<T>, KeySelector]>();
+        if (!this.tryAddAllProjectionMethods(events, methods, eventTypes, container, logger)) {
+            logger.warn(`Could not create projection ${this._projectionType.name} because it contains invalid projection methods`);
+            return;
+        }
+        const projection = new Projection<T>(decoratedType.projectionId, decoratedType.readModel, decoratedType.scopeId, events);
+        projections.register<T>(
+            new ProjectionProcessor<T>(
+                projection,
+                client,
+                executionContext,
+                eventTypes,
+                logger
+            ), cancellation);
+    }
+
+
+
+    private tryAddAllProjectionMethods(events: EventTypeMap<[ProjectionCallback<any>, KeySelector]>, methods: OnDecoratedMethod[], eventTypes: IEventTypes, container: IContainer, logger: Logger): boolean {
+        let allMethodsValid = true;
+        for (const method of methods) {
+            const [hasEventType, eventType] = this.tryGetEventTypeFromMethod(method, eventTypes);
+
+            if (!hasEventType) {
+                allMethodsValid = false;
+                logger.warn(`Could not create projection method ${method.name} in projection ${this._projectionType.name} because it is not associated to an event type`);
+                continue;
+            }
+
+            const onMethod = this.createProjectionMethod(method, container);
+            const keySelector = method.keySelector;
+
+            if (events.has(eventType!)) {
+                allMethodsValid = false;
+                logger.warn(`Event handler ${this._projectionType.name} has multiple projection methods handling event type ${eventType}`);
+                continue;
+            }
+            events.set(eventType!, [onMethod, keySelector]);
+        }
+        return allMethodsValid;
+    }
+
+    private createProjectionMethod(method: OnDecoratedMethod, container: IContainer): ProjectionCallback<any> {
+        return (instance, event, projectionContext) => {
+            const result = method.method.call(instance, event, projectionContext);
+            if (result instanceof DeleteReadModelInstance) {
+                return result;
+            }
+            return instance;
+        };
+    }
```

Comment:
Don't think we need the container here - see comment further up about DI in projections.

Reply:
Removed it per earlier comment